pkg/controller/installation/products/rhsso/reconciler.go

```diff
@@ -0,0 +1,190 @@
+package rhsso
+
+import (
+	"context"
+	"errors"
+	aerogearv1 "github.com/integr8ly/integreatly-operator/pkg/apis/aerogear/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/pkg/controller/installation/marketplace"
+	"github.com/integr8ly/integreatly-operator/pkg/controller/installation/products/config"
+	coreosv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"k8s.io/client-go/kubernetes"
+
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	pkgclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+)
+
+var (
+	defaultInstallationNamespace = "rhsso"
+	keycloakName                 = "rhsso"
+)
+
+func NewReconciler(client pkgclient.Client, serverClient pkgclient.Client, coreClient *kubernetes.Clientset, configManager config.ConfigReadWriter, instance *v1alpha1.Installation, clusterHasOLM bool) (*Reconciler, error) {
+	config, err := configManager.ReadRHSSO()
+	if err != nil {
+		return nil, err
+	}
+	if config.GetNamespace() == "" {
+		config.SetNamespace(instance.Spec.NamespacePrefix + defaultInstallationNamespace)
+	}
+	var mpm marketplace.MarketplaceInterface
+	if clusterHasOLM {
```

Comment:
as we are not considering 3.x any more (for component installs), we can remove this if statement and assume it is always OLM.

Reply:
:+1:

Reply:
https://github.com/integr8ly/integreatly-operator/pull/18/commits/90a12657dfd39c6569608da602134dcebf73d002